### PR TITLE
second part of "buildslave" to "buildbot-worker" transition

### PR DESCRIPTION
**Pull request by rutsky**
Rename of "slave" stuff to "worker" in buildbot_worker using tools such as PyCharm's refactor.

Currently base on bug2340 branch, which should be merged into master soon.

Changes are done using refactoring tools, except for the following commits: 
b496e0364db905fbb5b7add6ac9fe341c80b35d8
46290de34aff990bf6f8df33b8e4ece1fec98217
79fe60465765bbae6857bb668d96494cefe6c2d6
53d76c6a7976e0108dd4b517552bc98bea95134b
64c4e8d3e4e4d9b0e49b1bb9cdef4baee5efcf80
d04ebbadb035fdfa5c3efff830618da6d0f7a42c

To review these changes should be enough to see commit log and take a look at commits with manual changes listed above.

### CODE REVIEW

**Comment by tardyp** [comment]
could you please rebase against part 1?

**Comment by tardyp** [comment]
github is dumb and wont notice I partially merged this PR. I would expect part2 will be nowreviewable after your rebase

**Comment by rutsky** [comment]
@tardyp done!

This PR is not finished yet, I think I have few more things that I can easily rename.
I'll mark it with appropriate tags when finish.

**Comment by rutsky** [comment]
@tardyp, ready for review!

File: worker/buildbot_worker/null.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def startService(self):
         # requires buildot version >= 0.9.0b5
         from buildbot.worker.protocols.null import Connection
 
-        yield BuildSlaveBase.startService(self)
+        yield WorkerBase.startService(self)
         # TODO: This is a workaround for using worker with "slave"-api with
```
Comment (tardyp) [inline worker/buildbot_worker/null.py]: slave in comment

Comment (tardyp) [inline worker/buildbot_worker/null.py]: forget about it

File: worker/buildbot_worker/scripts/create_worker.py [hunk 2]
```diff
@@ -32,11 +32,11 @@
     import os.path
     basedir = os.path.abspath(os.path.dirname(__file__))
 
-# note: this line is matched against to check that this is a buildslave
+# note: this line is matched against to check that this is a worker
 # directory; do not edit it.
 application = service.Application('buildslave')
```
Comment (tardyp) [inline worker/buildbot_worker/scripts/create_worker.py]: shall we also change that one? I guess so

File: worker/buildbot_worker/scripts/base.py [hunk 3]
```diff
@@ -18,9 +18,9 @@
...
 
 
-def isBuildslaveDir(dir):
+def isWorkerDir(dir):
     def print_error(error_message):
-        log.msg("%s\ninvalid buildslave directory '%s'" % (error_message, dir))
+        log.msg("%s\ninvalid worker directory '%s'" % (error_message, dir))
 
     buildbot_tac = os.path.join(dir, "buildbot.tac")
     try:
```
Comment (tardyp) [inline worker/buildbot_worker/scripts/base.py]: should change 
"Application('buildslave')

a few lines after (gh wont allow to comment there)

File: worker/buildbot_worker/scripts/create_worker.py [hunk 4]
```diff
@@ -209,21 +209,21 @@ def createSlave(config):
         config['allow-shutdown'] = repr(asd)
 
     if config['no-logrotate']:
-        slaveTAC = "".join([slaveTACTemplate[0]] + slaveTACTemplate[2:])
+        workerTAC = "".join([workerTACTemplate[0]] + workerTACTemplate[2:])
     else:
```
Comment (tardyp) [inline worker/buildbot_worker/scripts/create_worker.py]: gosh thats ugly..

File: worker/buildbot_worker/scripts/create_worker.py (hunk 2 above)
Comment (rutsky) [inline worker/buildbot_worker/scripts/create_worker.py]: Yes, this should be changed.
I want to get in this PR most of semantically insignificant changes, for which no careful review is needed.

This particular change requires thinking IMO and I would like to do it in separate PR.

@tardyp, also there are bunch of not renamed strings that I think contains errors, I didn't touched them too.

File: worker/buildbot_worker/test/unit/test_scripts_upgrade_slave.py [hunk 5]
```diff
@@ -35,13 +35,13 @@
 """
 
 
-class TestUpgradeSlave(misc.IsBuildslaveDirMixin,
-                       misc.FileIOMixin,
-                       misc.LoggingMixin,
-                       unittest.TestCase):
+class TestUpgradeWorker(misc.IsWorkerDirMixin,
```
Comment (tardyp) [inline worker/buildbot_worker/test/unit/test_scripts_upgrade_slave.py]: I'd guess this will not really work anymore. will it?

File: worker/buildbot_worker/scripts/upgrade_worker.py [hunk 6]
```diff
@@ -20,10 +20,10 @@
...
 
-def upgradeSlave(config):
+def upgradeWorker(config):
     basedir = os.path.expanduser(config['basedir'])
 
-    if not base.isBuildslaveDir(basedir):
+    if not base.isWorkerDir(basedir):
         return 1
 
     buildbot_tac = open(os.path.join(basedir, "buildbot.tac")).read()
```
Comment (tardyp) [inline worker/buildbot_worker/scripts/upgrade_worker.py]: I'm not sure we should even bother to keep this command @djmitche opinion?

File: worker/buildbot_worker/test/unit/test_scripts_upgrade_slave.py (hunk 5 above)
Comment (rutsky) [inline worker/buildbot_worker/test/unit/test_scripts_upgrade_slave.py]: This test will work :) However whole upgrade step will be removed, since `buildbot_worker` will not have any versions (yet) to upgrade from. I'll do it in separate PR.

**Comment by tardyp** [comment]
looks good so far. a few comments to address on a later stage

File: worker/buildbot_worker/scripts/upgrade_worker.py (hunk 6 above)
Comment (rutsky) [inline worker/buildbot_worker/scripts/upgrade_worker.py]: IMO, `upgrade-worker` *may* not be completely dropped from sources (since perhaps at some point we will need such `upgrade-worker`), but for current version this command at least should be disabled, removed from help message, removed from docs, etc.

But I would prefer to drop it completely — less code, less to maintain.

**Comment by rutsky** [comment]
@tardyp I replied to your comments. In short: I don't want to include any semantically significant changes in this PR, rather lets do them in separate PRs that will be easier to accurately review.

**Comment by djmitche** [comment]
Yes, please drop it.. over a decade and we've never needed it..

**Comment by djmitche** [comment]
A few comments, but no serious concerns here.

**Comment by rutsky** [comment]
@djmitche, @tardyp, is all issues resolved with this PR? I think I answered on all comments.
